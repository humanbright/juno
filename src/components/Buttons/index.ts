export * from './Button';
export * from './ButtonBar';
export * from './ButtonBase';
export * from './ButtonGroup';
export * from './FabButton';
export * from './FabIconButton';
export * from './IconButton';
export * from './IconButtonGroup';
export * from './RoundButton';
export * from './SplitButton';
export * from './ToggleButton';
export * from './ToggleButtonGroup';
