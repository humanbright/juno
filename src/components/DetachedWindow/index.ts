export * from './DetachedWindow';
export * from './DetachedWindowStylesProvider';
