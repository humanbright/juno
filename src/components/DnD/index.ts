export * from './DragDropContext';
export * from './Draggable';
export * from './Droppable';
export * from './DragHandle';
