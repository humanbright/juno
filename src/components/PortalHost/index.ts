export * from './Connectable';
export * from './MultiplePortalHost';
export * from './PortalControllerProvider';
export * from './PortalHost';
export * from './PortalManager';
