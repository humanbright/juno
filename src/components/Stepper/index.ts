export * from './Step';
export * from './StepButton';
export * from './StepConnector';
export * from './StepContent';
export * from './StepIcon';
export * from './StepLabel';
export * from './Stepper';
