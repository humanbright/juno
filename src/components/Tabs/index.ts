export * from './Tabs';
export * from './Tab';
export * from './TabContext';
export * from './TabList';
export * from './TabPanel';
