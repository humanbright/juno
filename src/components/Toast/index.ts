export * from './Toast';
export * from './ToastContainer';
export { ToastContainerStyle } from './styles';
