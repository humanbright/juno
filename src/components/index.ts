export * from './Accordion';
export * from './Alert';
export * from './Animations';
export * from './AppBar';
export * from './Avatar';
export * from './Backdrop';
export * from './Badge';
export * from './Box';
export * from './Buttons';
export * from './Card';
export * from './Chip';
export * from './ClickAwayListener';
export * from './DetachedWindow';
export * from './Dialer';
export * from './Dialog';
export * from './Divider';
export * from './DnD';
export * from './Downshift';
export * from './Drawer';
export * from './Forms';
export * from './Grid';
export * from './Hidden';
export * from './Icon';
export * from './ImageView';
export * from './InlineEditable';
export * from './Link';
export * from './List';
export * from './Loading';
export * from './Menu';
export * from './Pagination';
export * from './Paper';
export * from './Popover';
export * from './Popper';
export * from './PopupBox';
export * from './Portal';
export * from './PortalHost';
export * from './Presence';
export * from './Progress';
export * from './Rating';
export * from './Responsive';
export * from './Snackbar';
export * from './Stepper';
export * from './TablePagination';
export * from './Tables';
export * from './Tabs';
export * from './Tag';
export * from './Text';
export * from './Thumbnail';
export * from './Toast';
export * from './Toolbar';
export * from './Tooltip';
export * from './Transitions';
export * from './Typography';
export * from './VirtualizedMenu';
export * from './Virtuoso';
export * from './VisuallyHidden';
