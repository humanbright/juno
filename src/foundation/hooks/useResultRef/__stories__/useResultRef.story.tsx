import React, { useEffect } from 'react';

import { Meta, Story } from '@storybook/react';

import { Title } from '../../../../storybook/components';
import { useResultRef } from '../useResultRef';

export default {
  title: 'ðŸ”§ Foundation/Hooks/useResultRef',
  argTypes: {},
} as Meta;

export const useResultRefExample: Story<{}> = () => {
  const ref = useResultRef(() => {
    console.log('only trigger once');
    return 1 + 1;
  });

  useEffect(() => {
    console.log('refA', ref.current); // get that current div dom
  }, []);

  return <Title>{ref.current}</Title>;
};

useResultRefExample.args = {};
useResultRefExample.storyName = 'useResultRef';
