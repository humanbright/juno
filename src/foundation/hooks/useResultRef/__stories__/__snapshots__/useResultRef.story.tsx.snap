// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ðŸ”§ Foundation/Hooks/useResultRef useResultRef 1`] = `
.c0 {
  font-size: 1.5rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 32px;
  color: #121212;
}

<h5
  className="MuiTypography-root c0 MuiTypography-paragraph MuiTypography-displayBlock"
  data-color="neutral.f06"
  data-variant="headline1"
>
  2
</h5>
`;
